astropy/units/structured.py

```diff
@@ -297,9 +297,10 @@ def cgs(self):
         return self._recursively_apply(operator.attrgetter("cgs"))
 
     # Needed to pass through Unit initializer, so might as well use it.
-    def _get_physical_type_id(self):
+    @property
```

Comment:
This might also be a useful place for a `cached_property` - the construction of the structure takes quite a bit of time.

Reply:
Should we cache the physical type ids of function units too?

Reply:
I wondered about that, but thought that in the end all one saves there is an extra attribute access. Though perhaps for consistency it is better to do it for function units too; in a way, it avoids our future selves thinking why one is cached and the other is not.